resources/api/reservation.py

```diff
@@ -87,15 +87,27 @@ def __init__(self, *args, **kwargs):
             if isinstance(self.instance, Reservation) and isinstance(self.instance.resource, Resource):
                 resource = self.instance.resource
 
-        # set extra fields required if the related resource is found and it needs manual confirmation
-        if resource and resource.need_manual_confirmation:
-            for field_name in RESERVATION_EXTRA_FIELDS:
+        # set supported and required extra fields
+        if resource:
+            required = set()
+            if resource.need_manual_confirmation:
+
+                # manually confirmed reservations have some extra conditions
+                can_approve = resource.can_approve_reservations(self.context['request'].user)
+                if data.get('staff_event', False) and can_approve:
+                    required = {'reserver_name', 'event_description'}
+
+            supported = resource.get_supported_reservation_extra_field_names()
+            if not required:
```

Comment:
I think it's clearer to move this right after `if resource:` and lose the `if`. Override `required` then as needed.